### ISSUE TITLE
[BUG] Whisker annotations misplaced on categorical axes

### ISSUE THREAD

**Issue by sggaffney** [bokeh/bokeh#10593]
#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
- conda environment: `conda create -n bk220 pandas bokeh=2.2.0`
  - python=3.6.10=hc70fcce_1
  - bokeh=2.2.0=py_0
- Chrome Version 86.0.4240.80
- Mac OSX 10.15.6

#### Description of expected behavior and the observed behavior

- Whisker positioning on categorical axes worked intuitively in Version 2.1.1, behaving in the same way (i.e. using the same coordinate system) as scatter positioning. Numeric string factors are interpreted as strings.
- Version 2.2.0 onwards has unexpected behavior: whisker locations don't respect factors, and numeric string factors behave differently to non-numeric strings.

#### Complete, minimal, self-contained example code that reproduces the issue

```python
from bokeh import io as bkio
from bokeh import models as bkm
from bokeh import plotting as bkp
from bokeh import layouts as bkl

import pandas as pd

df = pd.DataFrame.from_records([
    {'n': '3', 'alpha': 'a3', 'name': 'three', 'low': 5, 'high': 10},
    {'n': '2', 'alpha': 'a2', 'name': 'two', 'low': 2, 'high': 4},
    {'n': '1', 'alpha': 'a1', 'name': 'one', 'low': 1, 'high': 2},
])

n_tuples = tuple(df['n'].values)
n_name_tuples = [tuple(i) for i in df[['n', 'name']].values]
alpha_name_tuples = [tuple(i) for i in df[['alpha', 'name']].values]

plots = []
for base, y_vals in [('n', n_tuples), ('n', n_name_tuples), ('alpha', alpha_name_tuples)]:
    source = bkm.ColumnDataSource(df)
    y_range = bkm.FactorRange(factors=y_vals)
    p = bkp.figure(y_range=y_range,  x_range=(0, 11), plot_width=200, plot_height=200)
    p.add_layout(bkm.Whisker(source=source, base=base, dimension='width',
                         upper="high", lower="low"))
    p.scatter(x='low', y=base, size=10, alpha=0.5, source=source)
    plots.append(p)

row = bkl.row(plots)
bkio.show(row)
```

#### Screenshots or screencasts of the bug in action
2.1.1 behavior
<img width="617" alt="whiskers_2 1 1" src="https://user-images.githubusercontent.com/5185412/96896302-41ad1480-1485-11eb-927d-09f8180d0af6.png">

2.2.0 (- 2.2.3) behavior
<img width="608" alt="whiskers_2 2 0" src="https://user-images.githubusercontent.com/5185412/96896438-686b4b00-1485-11eb-9bd9-d09a97925e31.png">

**Comment by bryevdv** [comment]
IIRC @mattpap you mentioned on the Discourse that this is already fixed in `branch-2.3`? 

* Was there ever an original issue to track this, linked to a PR?
* If not, do you know offhand the root cause and what fixed it in the current branch?
* Is there currently a test that cover this regression in the future?

**Comment by mattpap** [comment]
Yesterday I was referring to an unrelated issue. This is a duplicate of issue #10575, fixed in PR #10577 (2.3.0dev2).
